File: sarasvant/lib/models/questions/msq.dart
import 'package:sarasvant/models/variable.dart';

class MSQ {
  String id;
  String bankId;
  String question;
  List<Variable> variable;
  int points;
  List<String> options;
  List<int> answerIndices;

  MSQ({
    required this.id,
    required this.bankId,
    required this.question,
    required this.variable,
    required this.points,
    required this.options,
    required this.answerIndices,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'bankId': bankId,
      'question': question,
      'variable': variable,
      'points': points,
      'options': options,
      'answerIndices': answerIndices,
    };
  }

  factory MSQ.fromJson(Map<String, dynamic> json) {
    return MSQ(
      id: json['id'],
      bankId: json['bankId'],
      question: json['question'],
      variable: json['variable'],
      points: json['points'],
      options: json['options'],
      answerIndices: json['answerIndices'],
    );
  }
}

File: server/internal/model/questions/msq.go
package questions

import "server/internal/model"

type MSQ struct {
	ID            string           `json:"id" firestore:"id"`
	BankID        string           `json:"bankId" firestore:"bankId"`
	Question      string           `json:"question" firestore:"question"`
	Variable      []model.Variable `json:"variable" firestore:"variable"`
	Points        int              `json:"points" firestore:"points"`
	Options       []string         `json:"options" firestore:"options"`
	AnswerIndices []int            `json:"answerIndices" firestore:"answerIndices"`
}
